src/m365/spo/commands/hubsite/hubsite-get.ts

```diff
@@ -25,42 +26,91 @@ class SpoHubSiteGetCommand extends SpoCommand {
     return 'Gets information about the specified hub site';
   }
 
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.url = typeof args.options.url !== 'undefined';
+    return telemetryProps;
+  }
+
   public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
     spo
       .getSpoUrl(logger, this.debug)
       .then((spoUrl: string): Promise<any> => {
-        const requestOptions: any = {
-          url: `${spoUrl}/_api/hubsites/getbyid('${encodeURIComponent(args.options.id)}')`,
-          headers: {
-            accept: 'application/json;odata=nometadata'
-          },
-          responseType: 'json'
-        };
-
-        return request.get(requestOptions);
+        return this.getHubSite(spoUrl, args.options);
       })
       .then((res: any): void => {
         logger.log(res);
         cb();
       }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
   }
 
+  private getHubSite(spoUrl: string, options: Options): Promise<HubSite> {
+    const requestOptions: any = {
+      url: `${spoUrl}/_api/hubsites`,
+      headers: {
+        accept: 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get(requestOptions)
+      .then((response: any) => {
+        let hubSites = response.value as HubSite[];
+
+        if (options.id) {
+          hubSites = hubSites.filter(site => site.ID.toLocaleLowerCase() === options.id!.toLocaleLowerCase());
+        }
+        else if (options.title) {
+          hubSites = hubSites.filter(site => site.Title.toLocaleLowerCase() === options.title!.toLocaleLowerCase());
+        }
+        else if (options.url) {
+          hubSites = hubSites.filter(site => site.SiteUrl.toLocaleLowerCase() === options.url!.toLocaleLowerCase());
+        }
+
+        if (!hubSites.length) {
+          return Promise.reject(`The specified hubsite ${options.id || options.title || options.url} does not exist`);
+        }
+
+        if (hubSites.length > 1) {
+          return Promise.reject(`Multiple hubsites with ${options.id || options.title || options.url} found: ${hubSites.map(site => site.SiteUrl)}`);
+        }
+
+        return hubSites[0];
```

Comment:
There shoud be no need for an additional conditional check at the end. Al the previous conditions would filter and check the array towards a single result list.

Reply:
sorry if maybe my comment wasn't clear. My idea was to use `find()` instead of `filter()` and always return the first found occurrence by condition

Reply:
If we use `find()` then we won't be able to return all the available options when we find more with the same title for example.

``` typescript
if (hubSites.length > 1) {
  return Promise.reject(`Multiple hubsites with ${options.id || options.title || options.url} found: ${hubSites.map(site => site.SiteUrl)}`);
}
```

Reply:
not sure if returning list of found hubsites with reject is what we would normally expect for a `get`  command 🤔. I would assume `get` always returns single item and `list` a collection of items. I think we may consider returning first hub site found by name or url. 
I also wonder how many hubs with the same url (or even title) would be on the same tenant 🤔. I would consider this a very special case to have two hub sites with the same url on the same tenant, also two hubs with the same title would be very misleading for the users 🤔. So I suggest it's safe to assume that it in most cases its ok to return the first found item by title or url.
I understand your point to be on the save side 👍. I just wonder if this not an overkill here 🤔 
@pnp/cli-for-microsoft-365-maintainers any opinion on this one ? TBH I think what is done now is not a mistake, just wondering if using `find()` will be good enough 🤔

Reply:
I agree with your statements. The approach used here is currently consistent with other newly created/updated commands. For example when we take a look at `m365 planner plan get` we get the following result:

![afbeelding](https://user-images.githubusercontent.com/38426621/174497260-93bfb9e4-d77d-4bbf-8394-449376c4cbdc.png)

This command also throws a reject when the plan can't be found. Still, you quote good points to why this should be approached differently. Maybe it's an idea to open a new issue so we could discusses this further and possible apply it to multiple commands. @Adam-it, what do you think? 😄

Reply:
I think in the newly created command it makes a bit more sense 🤔. I mean it's more possible to have plans or buckets with same name. For hubsites would be really special case for sure confusing for users and if anyone would like to list all hubsites to check what's going on then she or he may for sure use list command to check all 🤔. That's what I would do.
But still I, as I said, I also agree with this approach, just not sure if this is not an overkill in this case 😉. 
lets wait for comment from someone else 👍

Reply:
I agree with the current implementation. If we expect to retrieve a hubsite by title but for some reason find multiple, if we just returned the first match it could lead to undesired outcomes like deleting the wrong site. Failing the command with a prompt to disambiguate by ID is the right way forward.

Reply:
Ok then, let's leave it as is 🙂. Thanks for the group discussion on this 👍.
Still using same title seems a very extreme case to me 😉 and for sure would give more problems, but since that's possible I think it's ok to have it like this.
For the url is not possible, but let's leave it to be consistent 👍